vm/src/builtins/type.rs

```diff
@@ -32,6 +34,28 @@ pub struct PyType {
     pub slots: PyTypeSlots,
 }
 
+#[repr(C)]
+pub struct PyHeapType {
+    pub typ: PyType,
+    pub number_methods: PyRwLock<PyNumberMethods>,
+}
+
+impl From<PyHeapType> for PyType {
+    fn from(heaptype: PyHeapType) -> Self {
+        //TODO: SAFYTY
```

Comment:
minor nitpick: spelling

```suggestion
        //TODO: SAFETY
```